doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -865,6 +865,30 @@ Controlling display
    interpreted from left to right, so in case of an overlap, the flags on the
    right have higher priority, meaning that `A,-A` is equivalent to `-A`.
 
+.. opt:: Debug "{+, @ident {? = @natural } }"
```

Comment:
Also, it's strange for `Warnings` and `Debug` to take a string, which then has to parsed with custom code rather than parsing that with the usual, well-tested main parser through productions in an mlg file.  It would be an additional rule for `Set`.  Please give some thought to this.

Reply:
It's a very common pattern though. A significant part of the string options (`Bullet Behavior`, `Default Goal Selector`, `Default Proof Mode`, `Default Proof Using`, `Diffs`, `Loose Hint Behavior` and of course `Warnings`) interpret their string argument. It's also the case for one attribute (`using`).

Reply:
How common is it? Except for `Default Goal Selector`, `Default Proof Using` and `Warnings`, the options you list choose a string from a small list of alternatives, for which a parser is hardly needed.  The first 3 would benefit from using the parser more directly.

The doc for `Default Goal Selector` doesn't match the code.  See `GoaL_select.parse_goal_selector`.  `Warnings` has its own custom parser.  `Default Proof Using` calls the main parser internally.

There is already one custom `Set` entry:
| "Set" "Firstorder" "Solver" ltac_expr

And there is this, too:
| "Debug" [ "On" | "Off" ]

`Debug` could at least use the main parser internally.

Reply:
One advantage of having a custom (simpler) parser is that it can also be used for command line arguments.

Reply:
I don't see the advantage.  You can easily parse a string corresponding to any nonterminal.